File: server/src/main/java/com/fptmarket/service/ProductService.java
package com.fptmarket.service;

import com.fptmarket.dto.request.ProductRequest;
import com.fptmarket.dto.request.RejectProductRequest;
import com.fptmarket.dto.response.ProductResponse;
import com.fptmarket.entity.ProductStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface ProductService {
    Page<ProductResponse> getAllApprovedProducts(
            String keyword, Long categoryId, Double minPrice, Double maxPrice, String condition, Pageable pageable);
    
    ProductResponse getProductBySlug(String slug);
    
    ProductResponse createProduct(ProductRequest request, List<MultipartFile> images);
    
    ProductResponse updateProduct(Long id, ProductRequest request, List<MultipartFile> images);
    
    void deleteProduct(Long id);
    
    Page<ProductResponse> getMyProducts(Pageable pageable);
    
    // Admin methods
    Page<ProductResponse> getAllProductsForAdmin(ProductStatus status, Pageable pageable);
    
    ProductResponse approveProduct(Long id);
    
    ProductResponse rejectProduct(Long id, RejectProductRequest request);
    
    ProductResponse hideProduct(Long id);
}

File: client/src/services/productService.js
import api from './api';

const productService = {
    getProducts: async (params) => {
        const response = await api.get('/api/products', { params });
        return response.data;
    },

    getProductBySlug: async (slug) => {
        const response = await api.get(`/api/products/${slug}`);
        return response.data;
    },

    getProductById: async (id) => {
        const response = await api.get(`/api/products/id/${id}`);
        return response.data;
    },

    createProduct: async (formData) => {
        const response = await api.post('/api/products', formData, {
            headers: {
                'Content-Type': 'multipart/form-data',
            },
        });
        return response.data;
    },

    updateProduct: async (id, formData) => {
        const response = await api.put(`/api/products/${id}`, formData, {
            headers: {
                'Content-Type': 'multipart/form-data',
            },
        });
        return response.data;
    },

    deleteProduct: async (id) => {
        const response = await api.delete(`/api/products/${id}`);
        return response.data;
    },

    getMyProducts: async (params) => {
        const response = await api.get('/api/products/me', { params });
        return response.data;
    }
};

export default productService;
